libsolidity/formal/SMTEncoder.cpp

```diff
@@ -2643,16 +2643,23 @@ pair<FunctionDefinition const*, ContractDefinition const*> SMTEncoder::functionC
 		auto funDef = dynamic_cast<FunctionDefinition const*>(_ref->annotation().referencedDeclaration);
 		if (!funDef)
 			return {funDef, _contract};
-		auto contextContract = _contract;
+		ContractDefinition const* contextContract = nullptr;
 		if (lookup == VirtualLookup::Virtual)
+		{
 			funDef = &funDef->resolveVirtual(*_contract);
+			contextContract = _contract;
+		}
 		else if (lookup == VirtualLookup::Super)
 		{
 			auto super = _contract->superContract(*_contract);
 			solAssert(super, "Super contract not available.");
 			funDef = &funDef->resolveVirtual(*_contract, super);
 			contextContract = super;
 		}
+		else if (lookup == VirtualLookup::Static)
+			contextContract = funDef->annotation().contract;
+		else
+			solAssert(false, "");
```

Comment:
Since `lookup` is of type `VirtualLookup` which is an enum class of three elements `Virtual`, `Super`, and `Static`, this assertion is redundant since all elements are covered. Could also be changed to a switch case, so the c++ compiler errors on unhandled cases if any.

Reply:
Done